src/contracts/core/DelegationManager.sol

```diff
@@ -249,18 +275,23 @@ contract DelegationManager is
             withdrawalRoots = new bytes32[](0);
         } else {
             withdrawalRoots = new bytes32[](strategies.length);
+            uint64[] memory totalMagnitudes = allocationManager.getTotalMagnitudes(operator, strategies);
+
             for (uint256 i = 0; i < strategies.length; i++) {
                 IStrategy[] memory singleStrategy = new IStrategy[](1);
                 uint256[] memory singleShare = new uint256[](1);
+                uint64[] memory singleTotalMagnitude = new uint64[](1);
                 singleStrategy[0] = strategies[i];
                 singleShare[0] = shares[i];
+                singleTotalMagnitude[0] = totalMagnitudes[i];
 
                 withdrawalRoots[i] = _removeSharesAndQueueWithdrawal({
                     staker: staker,
                     operator: operator,
                     withdrawer: staker,
                     strategies: singleStrategy,
-                    shares: singleShare
+                    sharesToWithdraw: singleShare,
+                    totalMagnitudes: singleTotalMagnitude
                 });
             }
```

Comment:
we should set the stakeMagnitudeBack to `PRECISION_FACTOR` upon undelegation

Reply:
do you mean the `stakerScalingFactor`? I realize it needs to be initialized to `PRECISION_FACTOR` either on delegation or on undelegation.

Reply:
Actually no you're correct, the staker can redeposit as shares with a delegated operator and their staker scaling factor needs to be re-initialized to 1

Reply:
will that not be handled by the calculate function though?

Reply:
resolved offline. undelegated stakers need scaling factor of 1e18